Userland/Libraries/LibCore/AnonymousBuffer.cpp

```diff
@@ -22,8 +23,11 @@ ErrorOr<AnonymousBuffer> AnonymousBuffer::create_with_size(size_t size)
 ErrorOr<NonnullRefPtr<AnonymousBufferImpl>> AnonymousBufferImpl::create(int fd, size_t size)
 {
     auto* data = mmap(nullptr, round_up_to_power_of_two(size, PAGE_SIZE), PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0);
-    if (data == MAP_FAILED)
-        return Error::from_errno(errno);
+    if (data == MAP_FAILED) {
+        int const saved_errno = errno;
+        (void)close(fd);
```

Comment:
~AnonymousBufferImpl does `VERIFY(rc == 0);` for this `close()`; maybe this here could do that too?

Reply:
I thought user may provide invalid/made up fd, and we don't want to crush because of this?
And sorry I didn't respond earlier, I didn't notice the comment :^).